ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot

```diff
@@ -275,34 +275,48 @@ Wait For DSCInitialization CustomResource To Be Ready
 
 Apply DataScienceCluster CustomResource
     [Documentation]
-    [Arguments]        ${dsc_name}=${DSC_NAME}
+    [Arguments]        ${dsc_name}=${DSC_NAME}      ${custom}='False'       ${custom_cmp}=''
     ${file_path} =    Set Variable    tasks/Resources/Files/
-    Log to Console    Requested Configuration:
-    FOR    ${cmp}    IN    @{COMPONENT_LIST}
-         TRY
-               Log To Console    ${cmp} - ${COMPONENTS.${cmp}}
-         EXCEPT
-               Log To Console    ${cmp} - Removed
-         END
-    END
-    Log to Console    message=Creating DataScience Cluster using yml template
-    Create DataScienceCluster CustomResource Using Test Variables
-    Apply Custom Manifest in DataScienceCluster CustomResource Using Test Variables
-    ${yml} =    Get File    ${file_path}dsc_apply.yml
-    Log To Console    Applying DSC yaml
-    Log To Console    ${yml}
-    ${return_code}    ${output} =    Run And Return Rc And Output    oc apply -f ${file_path}dsc_apply.yml
-    Log To Console    ${output}
-    Should Be Equal As Integers  ${return_code}  0  msg=Error detected while applying DSC CR
-    Remove File    ${file_path}dsc_apply.yml
-    FOR    ${cmp}    IN    @{COMPONENT_LIST}
-           IF    $cmp not in $COMPONENTS
-                Component Should Not Be Enabled    ${cmp}
-           ELSE IF    '${COMPONENTS.${cmp}}' == 'Managed'
-                Component Should Be Enabled    ${cmp}
-           ELSE IF    '${COMPONENTS.${cmp}}' == 'Removed'
-                Component Should Not Be Enabled    ${cmp}
-           END
+    IF      ${custom} == True
+        Log to Console    message=Creating DataScience Cluster using custom configuration
+        Create DataScienceCluster CustomResource Using Custom Configuration
+        Apply Custom Manifest in DataScienceCluster CustomResource Using Custom Configuration
+        ${yml} =    Get File    ${file_path}dsc_apply.yml
+        Log To Console    Applying DSC yaml
+        Log To Console    ${yml}
+        ${return_code}    ${output} =    Run And Return Rc And Output    oc apply -f ${file_path}dsc_apply.yml
```

Comment:
is this applying twice the DSC? at line 282 (maybe in 283) and 287?

Reply:
Ah I see, the keyword says "Apply" but it doesn-t mean to actually Apply in the cluster via OC. Unless it's only me seeing the ambiguity, would you mind clear it a bit?

Reply:
That part follows the same logic as before (just some common parts) it is not newly added logic. On these lines it is just getting the file to log it into console

Reply:
yeah but still, the ambiguity reading the code remains